src/tools/illink/src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -1421,6 +1524,41 @@ internal static bool BindingFlagsAreUnsupported (BindingFlags? bindingFlags)
 			GetDynamicallyAccessedMemberTypesFromBindingFlagsForProperties (bindingFlags) |
 			GetDynamicallyAccessedMemberTypesFromBindingFlagsForNestedTypes (bindingFlags);
 
+		internal static bool HasMemberType (MemberTypes? memberTypes, MemberTypes? search) => memberTypes != null && (memberTypes & search) == search;
+
+		internal static DynamicallyAccessedMemberTypes GetDynamicallyAccessedMemberTypesFromMemberTypesForNestedTypes (MemberTypes? memberTypes) =>
+			(HasMemberType (memberTypes, MemberTypes.TypeInfo) ? DynamicallyAccessedMemberTypes.PublicNestedTypes | DynamicallyAccessedMemberTypes.NonPublicNestedTypes : DynamicallyAccessedMemberTypes.None) |
+			(HasMemberType (memberTypes, MemberTypes.NestedType) ? DynamicallyAccessedMemberTypes.PublicNestedTypes | DynamicallyAccessedMemberTypes.NonPublicNestedTypes : DynamicallyAccessedMemberTypes.None) |
```

Comment:
I'm slightly confused with the distinction between `TypeInfo` and `NestedType` - I think they both refer to nested types?
I assume `TypeInfo` would just be for when there's a nested type without `nested` attribute in IL? Otherwise, I have no idea what it is.
Can someone confirm what it means?